### ISSUE TITLE
rpiCam not found in v1.9.0  (Bookworm, libcamera0.3)

### ISSUE THREAD

**Issue by nathan-stewart** [bluenviron/mediamtx#3714]
## Which version are you using?
v1.9.0

## Which operating system are you using?
Raspberry PiOS Bookworm

- [ ] Linux amd64 standard
- [ ] Linux amd64 Docker
- [x ] Linux arm64 standard
- [ ] Linux arm64 Docker
- [ ] Linux arm7 standard
- [ ] Linux arm7 Docker
- [ ] Linux arm6 standard
- [ ] Linux arm6 Docker
- [ ] Windows amd64 standard
- [ ] Windows amd64 Docker (WSL backend)
- [ ] macOS amd64 standard
- [ ] macOS amd64 Docker
- [ ] Other (please describe)

## Describe the issue
2024/08/30 20:13:18 ERR [path front] [RPI Camera source] fork/exec /dev/shm/mediamtx-rpicamera-1725063198554484679/mtxrpicam: no such file or directory

## Describe how to replicate the issue
Raspberry Pi4 w/ Raspberry Pi OS (Bookworm).  I added the Pi Camera per the docs and it fails to find it. 
[code]
paths:
  front:
    source: rpiCamera
[/code]

I'm not using legacy camera interface,  vcgencmd get_camera returns :
    supported=0 detected=0, libcamera interfaces=0

rpicam-hello  finds it and returns:
[1:58:39.119744461] [2158]  INFO Camera camera_manager.cpp:313 libcamera v0.3.0+65-6ddd79b5
[1:58:39.160407618] [2164]  WARN RPiSdn sdn.cpp:40 Using legacy SDN tuning - please consider moving SDN inside rpi.denoise
[1:58:39.163170698] [2164]  INFO RPI vc4.cpp:446 Registered camera /base/soc/i2c0mux/i2c@1/ov5647@36 to Unicam device /dev/media2 and ISP device /dev/media0
[1:58:39.163284642] [2164]  INFO RPI pipeline_base.cpp:1104 Using configuration file '/usr/share/libcamera/pipeline/rpi/vc4/rpi_apps.yaml'
Preview window unavailable
Mode selection for 1296:972:12:P
    SGBRG10_CSI2P,640x480/0 - Score: 3296
    SGBRG10_CSI2P,1296x972/0 - Score: 1000
    SGBRG10_CSI2P,1920x1080/0 - Score: 1349.67
    SGBRG10_CSI2P,2592x1944/0 - Score: 1567
Stream configuration adjusted
[1:58:39.165205565] [2158]  INFO Camera camera.cpp:1183 configuring streams: (0) 1296x972-YUV420 (1) 1296x972-SGBRG10_CSI2P
[1:58:39.165592560] [2164]  INFO RPI vc4.cpp:621 Sensor: /base/soc/i2c0mux/i2c@1/ov5647@36 - Selected sensor format: 1296x972-SGBRG10_1X10 - Selected unicam format: 1296x972-pGAA
#0 (0.00 fps) exp 33222.00 ag 7.94 dg 1.01
...

Libcamera is: libcamera0.3:armhf                   0.3.0+rpt20240617-1  

1. start the server
2. publish with ...
3. read with ...

## Did you attach the server logs?
yes

[mediamtx-pi4-bookworm.log](https://github.com/user-attachments/files/16822904/mediamtx-pi4-bookworm.log)

## Did you attach a network dump?

 no

**Comment by matdehaast** [comment]
Also experiencing this issue on pi zero 2w.

**Comment by aler9** [comment]
this happens because you are using the 32-bit operative system and the 64-bit server. Use the same architecture on both the operating system and server, as explicitly mentioned in the README.

**Comment by nathan-stewart** [comment]
It's not so explicit in the README. It simply says "both are supported". While it should be obvious to use the one that matches the OS, since Dec 2023 Raspberry Pi Os ships with a 64 bit kernel even when installed 32 bit, and uname reports aarch64 in both cases. Even if the user is aware of that, it's not clear which the app needs (ie  if it says aarch64 you'd naturally want to suggest one built aarch64.)

**Comment by emannersvip** [comment]
I’m interested in this ticket. I downloaded bookworm 64 bit on a raspberry pi 5 with the pi camera 3.When I installed medianmtx 1.9.0 armv864 and I update the medianmtx.yml to use rpiSource with basic settings I get this error also.I’m at work but once I get home I can upload logs if that’s not considered hijacking this thread.Sent from my phoneOn Sep 3, 2024, at 5:50 AM, nathan-stewart ***@***.***> wrote:﻿
It's not so explicit in the README. It simply says "both are supported". While it should be obvious to use the one that matches the OS, since Dec 2023 Raspberry Pi Os ships with a 64 bit kernel even when installed 32 bit, and uname reports aarch64 in both cases. Even if the user is aware of that, it's not clear which the app needs (ie  if it says aarch64 you'd naturally want to suggest one built aarch64.)
